### PR TITLE
[mvvm] MVVM pattern

### PR DESCRIPTION
**Pull request by lvjian700**
## MVVM pattern

* https://objccn.io/issue-13-1/ 
* http://www.infoq.com/cn/articles/rethinking-mvc-mvvm
* http://www.ruanyifeng.com/blog/2015/02/mvcmvp_mvvm.html
* https://blog.coding.net/blog/ios-architecture-patterns 
* https://medium.com/ios-os-x-development/ios-architecture-patterns-ecba4c38de52

## Treat UIView as View

* Introduce `MomentViewModel` for `SingleMomentCell` 

## Treat UIViewController as View

* Introduce `MomentsViewModel` for `ViewController`
* Introduce `MomentRepository`

### CODE REVIEW

File: BabyMoment/MomentsViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+class MomentsViewModel {
+    var cellViewModels: [MomentViewModel]?
+
+    let repository:     MomentRepositoryProtocol
+    let birthday:       NSDate
```
Comment (iossocket) [inline BabyMoment/MomentsViewModel.swift:7]: Maybe could use Date, instead of NSDate.

File: BabyMoment/SingleMomentCell.swift [hunk 2]
```diff
@@ -11,61 +11,44 @@ class SingleMomentCell: UITableViewCell, UITextFieldDelegate {
...
-    
-    private func setDate(date: NSDate) {
-        let formatter = NSDateFormatter()
-        formatter.dateFormat = "dd"
-        day.text = formatter.stringFromDate(date)
+    func configureCell(cellViewModel: MomentViewModel) {
+        self.viewModel      = cellViewModel
+        self.textField.text = viewModel.message
+        self.time.text      = viewModel.photoTakenDesc
+        self.timeAgo.text   = viewModel.uploadedAtDesc
```
Comment (iossocket) [inline BabyMoment/SingleMomentCell.swift:39]: Do not need to align the `=`, ^_^

File: BabyMoment/SingleMomentCell.swift [hunk 3]
```diff
@@ -11,61 +11,44 @@ class SingleMomentCell: UITableViewCell, UITextFieldDelegate {
...
-    private func setUploadedAt(date: NSDate) {
-        let timeAgoText:String = date.timeAgoSinceNow()
-        timeAgo.text = timeAgoText
-    }
-}
-
-extension UITableViewCell {
     func setImageFromLocal(imageView: UIImageView, assetLocationId: String) {
         let result: PHFetchResult = PHAsset.fetchAssetsWithLocalIdentifiers([assetLocationId], options: nil)
+        if result.firstObject == nil {
```
Comment (iossocket) [inline BabyMoment/SingleMomentCell.swift]: `if let` or `guard let` is better.

File: BabyMoment/MomentRepository.swift [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import Foundation
+import RealmSwift
+
+class MomentRepository: MomentRepositoryProtocol {
+    var notificationToken: NotificationToken? = nil
+
+    func moments() -> [Moment] {
+        let realm = try! Realm()
```
Comment (iossocket) [inline BabyMoment/MomentRepository.swift]: try to avoid to use `!`

Comment (edxzh) [reply BabyMoment/MomentRepository.swift]: @iossocket that's very good idea. 💯

File: BabyMoment/MomentsViewModel.swift (hunk 1 above)
Comment (lvjian700) [reply BabyMoment/MomentsViewModel.swift:7]: It's `swift 2.3` 😿

**Pull request closed without merging**